packages/core/src/modules/proofs/protocol/v2/V2ProofService.ts

```diff
@@ -0,0 +1,808 @@
+import type { CreateProblemReportOptions } from '../../formats/models/ProofFormatServiceOptions'
+import type { ProofFormatSpec } from '../../formats/models/ProofFormatSpec'
+import type {
+  CreateAckOptions,
+  CreatePresentationOptions,
+  CreateProposalAsResponseOptions,
+  CreateProposalOptions,
+  CreateRequestAsResponseOptions,
+  CreateRequestOptions,
+  GetRequestedCredentialsForProofRequestOptions,
+  ProofRequestFromProposalOptions,
+} from '../../models/ProofServiceOptions'
+import type {
+  RetrievedCredentialOptions,
+  ProofRequestFormats,
+  RequestedCredentialsFormats,
+} from '../../models/SharedOptions'
+
+import { inject, Lifecycle, scoped } from 'tsyringe'
+
+import { AgentConfig } from '../../../../agent/AgentConfig'
+import { EventEmitter } from '../../../../agent/EventEmitter'
+import { InjectionSymbols } from '../../../../constants'
+import { AriesFrameworkError } from '../../../../error'
+import { DidCommMessageRepository, DidCommMessageRole } from '../../../../storage'
+import { Wallet } from '../../../../wallet/Wallet'
+import { AckStatus } from '../../../common'
+import { ConnectionService } from '../../../connections'
+import { ProofEventTypes } from '../../ProofEvents'
+import { ProofService } from '../../ProofService'
+import { ProofsUtils } from '../../ProofsUtil'
+import { PresentationProblemReportReason } from '../../errors/PresentationProblemReportReason'
+import { V2_INDY_PRESENTATION_PROPOSAL, V2_INDY_PRESENTATION_REQUEST } from '../../formats/ProofFormats'
+import { IndyProofFormatService } from '../../formats/indy/IndyProofFormatService'
+import { ProofProtocolVersion } from '../../models/ProofProtocolVersion'
+import { ProofState } from '../../models/ProofState'
+import { PresentationRecordType, ProofRecord, ProofRepository } from '../../repository'
+
+import { V2PresentationProblemReportError } from './errors'
+import { V2PresentationAckHandler } from './handlers/V2PresentationAckHandler'
+import { V2PresentationHandler } from './handlers/V2PresentationHandler'
+import { V2PresentationProblemReportHandler } from './handlers/V2PresentationProblemReportHandler'
+import { V2ProposePresentationHandler } from './handlers/V2ProposePresentationHandler'
+import { V2RequestPresentationHandler } from './handlers/V2RequestPresentationHandler'
+import { V2PresentationAckMessage } from './messages'
+import { V2PresentationMessage } from './messages/V2PresentationMessage'
+import { V2PresentationProblemReportMessage } from './messages/V2PresentationProblemReportMessage'
+import { V2ProposalPresentationMessage } from './messages/V2ProposalPresentationMessage'
+import { V2RequestPresentationMessage } from './messages/V2RequestPresentationMessage'
+
+@scoped(Lifecycle.ContainerScoped)
+export class V2ProofService extends ProofService {
+  public async createProofRequestFromProposal(options: ProofRequestFromProposalOptions): Promise<ProofRequestFormats> {
```

Comment:
I think this method should be placed below the properties and constructor (and grouped with proof request / proposal methods)